tasks/benchmark/run-benchmarks.mts

```diff
@@ -0,0 +1,225 @@
+  console.log(`${REDWOOD_PROJECT_DIRECTORY}\n`)
+
+  fs.mkdirSync(REDWOOD_PROJECT_DIRECTORY, { recursive: true })
+
+  // Register clean up
+  if (args.cleanUp) {
+    console.log('The directory will be deleted after the tests are run')
+    process.on('SIGINT', () => {
+      if(!cleanUpExecuted){
+        cleanUp({
+          projectPath: REDWOOD_PROJECT_DIRECTORY,
+        })
+        cleanUpExecuted = true
+      }
+    })
+    process.on('exit', () => {
+      if(!cleanUpExecuted){
+        cleanUp({
+          projectPath: REDWOOD_PROJECT_DIRECTORY,
+        })
+        cleanUpExecuted = true
+      }
+    })
+  }
+
+  // Get all the setups
+  const setups = fg
+    .sync('*', {
+      onlyDirectories: true,
+      cwd: SETUPS_DIR,
+    })
+    .filter((setupDir) => {
+      return (
+        args.setup.length === 0 ||
+        args.setup.some((setup) => setupDir.includes(setup.toString()))
+      )
+    })
+
+  if (setups.length === 0) {
+    console.log('\nThere are no setups to run.')
+    process.exit(0)
+  }
+
+  // Create a test project and sync the current framework state
+  console.log('\nCreating a fresh project:')
+
+  buildRedwoodFramework({
+    frameworkPath: REDWOODJS_FRAMEWORK_PATH,
+  })
+  createRedwoodJSApp({
+    frameworkPath: REDWOODJS_FRAMEWORK_PATH,
+    projectPath: REDWOOD_PROJECT_DIRECTORY,
+    typescript: true
+  })
+  addFrameworkDepsToProject({
+    frameworkPath: REDWOODJS_FRAMEWORK_PATH,
+    projectPath: REDWOOD_PROJECT_DIRECTORY,
+  })
+  runYarnInstall({
+    projectPath: REDWOOD_PROJECT_DIRECTORY,
+  })
+  copyFrameworkPackages({
+    frameworkPath: REDWOODJS_FRAMEWORK_PATH,
+    projectPath: REDWOOD_PROJECT_DIRECTORY,
+  })
+  initGit({
+    projectPath: REDWOOD_PROJECT_DIRECTORY,
+  })
+
+  // zx setup
+  $.verbose = false
+  $.cwd = REDWOOD_PROJECT_DIRECTORY
+  $.log = () => {}
+
+  console.log('\nThe following setups will be run:')
+  for(let i = 0; i < setups.length; i++){
+    console.log(`- ${setups[i]}`)
+  }
+
+  for (const setup of setups) {
+    // import the setup
+    const setupFile = path.join(SETUPS_DIR, setup, 'setup.mjs')
+    const setupModule = await import(setupFile)
+    const runForTests = args.test.length === 0 ?
+      setupModule.validForTests :
+      args.test.filter((test) => setupModule.validForTests.includes(test))
+
+    if(runForTests.length === 0){
+      console.log(`\nThere are no tests to run for setup ${setup}, skipping...`)
+      continue
+    }
+
+    // Clean up the project state
+    console.log(`\nCleaning up the project state...`)
+    await $`git reset --hard`
+    await $`git clean -fd`
+
+    // await execa('git', ['reset', '--hard'], {
+    //   cwd: REDWOOD_PROJECT_DIRECTORY,
+    //   stdio: args.verboseGit ? 'inherit' : 'ignore',
+    // })
+    // await execa('git', ['clean', '-fd'], {
+    //   cwd: REDWOOD_PROJECT_DIRECTORY,
+    //   stdio: args.verboseGit ? 'inherit' : 'ignore',
+    // })
```

Comment:
I had to switch to use zx for some steps - specifically for serving the api. When using execa I had major issues trying to get the process to exit on demand  - I tried for some time but lost my patience and switched to use zx for it

The left over commented code can go.
```suggestion
```